app/controllers/ingredient_controller.rb

```diff
@@ -0,0 +1,47 @@
+class IngredientController < ApplicationController
+  def new
+    if params[:name]
+      i = Ingredient.create(:name => params[:name])
+      if i.valid?
+        
+      else
+        render :status => 500
+      end
+    else
+      render :status => 500
+    end
+  end
+
+  def index
+    @ingredients = Ingredient.all
+  end
+
+  def update
+    if params[:id] and params[:name]
+      
+      if Ingredient.exists?(params[:id])
+        i = Ingredient.find(params[:id])
+        i.name = params[:name]
+        i.save
```

Comment:
```ruby
 +        i = Ingredient.find(params[:id])
 +        i.name = params[:name]
 +        i.save
```
 1 line instead of 3 
```ruby
Ingredient.find( params[:id] ).update( name: params[:name] )
```